NightscoutUploadKit/NightscoutUploader.swift

```diff
@@ -277,8 +298,69 @@ public class NightscoutUploader {
             completion(.failure(UploadError.invalidParameters))
         }
     }
+    
+    
+    // MARK: - Loop Notifications
+
+    public func startOverride(overrideName: String, reasonDisplay: String, durationInMinutes: Int) async throws {
+
+        let jsonDict: [String: String] = [
+            "reason": overrideName,
+            "reasonDisplay": reasonDisplay,
+            "eventType": "Temporary Override",
+            "duration": "\(durationInMinutes)",
+            "notes": ""
+        ]
+        
+        try await postNotification(payload: jsonDict)
+    }
+    
+    public func cancelOverride() async throws {
+
+        let jsonDict: [String: String] = [
+            "eventType": "Temporary Override Cancel",
+            "duration": "0"
+        ]
+        
+        try await postNotification(payload: jsonDict)
+    }
+    
+    public func deliverBolus(amountInUnits: Double, otp: String) async throws  {
 
+        let jsonDict: [String: String] = [
+            "eventType": "Remote Bolus Entry",
+            "remoteBolus": "\(amountInUnits)",
+            "otp": otp
+        ]
+        
+        try await postNotification(payload: jsonDict)
+    }
+    
+    public func deliverCarbs(amountInGrams: Double, absorptionInHours: Double, consumedDate: Date? = nil, otp: String) async throws  {
+
+        var jsonDict: [String: String] = [
+            "eventType": "Remote Carbs Entry",
+            "remoteCarbs": "\(amountInGrams)",
+            "remoteAbsorption": "\(absorptionInHours)",
+            "otp": otp
+        ]
+        
+        if let consumedDate {
+            let formatter = ISO8601DateFormatter()
+            formatter.formatOptions =  [.withInternetDateTime, .withFractionalSeconds]
+            jsonDict["created_at"] = formatter.string(from: consumedDate)
+        }
+        
+        try await postNotification(payload: jsonDict)
+    }
 
+    public func postNotification(payload: [String: String]) async throws  {
+        guard let url = url(for: .notifications) else {
+            throw UploadError.missingConfiguration
+        }
+        
```

Comment:
The notification API returns an "OK" string in the body. I _could_ check that but I feel the 200 response that gets checked in `postToNS` is sufficient.

Reply:
The http status should override any body text in terms of failure semantics, I'd posit.  If NS is returning errors in responses with a 200 status, that'd be a bug in NS, I'd think.